public/Set-DbaDbQueryStoreOption.ps1

```diff
@@ -170,7 +170,7 @@ function Set-DbaDbQueryStoreOption {
             }
 
             # We have to exclude all the system databases since they cannot have the Query Store feature enabled
-            $dbs = Get-DbaDatabase -SqlInstance $server -ExcludeDatabase $ExcludeDatabase -Database $Database | Where-Object IsAccessible
+            $dbs = Get-DbaDatabase -SqlInstance $server -ExcludeDatabase $ExcludeDatabase -Database $Database | Where-Object { ($_.IsAccessible) -and (!$_.IsDatabaseSnapshot) }
```

Comment:
does `{ $_.IsAccessible -and !$_.IsDatabaseSnapshot }` work, too ? If so, it's a little bit clearer.

Reply:
even better, remove the filter alltogether and use it like on https://github.com/dataplat/dbatools/blob/6610b1190e6dbffca49124ba7425c5ad95ed0ff6/public/Set-DbaDbQueryStoreOption.ps1#L178C17-L181C18 to signal the db is a snapshot, skipping it

Reply:
> does `{ $_.IsAccessible -and !$_.IsDatabaseSnapshot }` work, too ? If so, it's a little bit clearer.

It does - and it goes along with the conventions I usual ly follow too so not sure why I didn't here!

> even better, remove the filter alltogether and use it like on https://github.com/dataplat/dbatools/blob/6610b1190e6dbffca49124ba7425c5ad95ed0ff6/public/Set-DbaDbQueryStoreOption.ps1#L178C17-L181C18 to signal the db is a snapshot, skipping it

I'd resprectfully disagree this is better for this particular check. The existing check for `IsAccessible` makes sense - you'd want to make sure that something that was accessible when extracting the list of databases is still accessible now we want to change it. Snapshots are always snapshots though, they're not going to suddenly change into an actual database that we can configure.

Reply:
makes sense, thanks